Laravel Horizon container fails to build due to tokenizer error

### Description:
When building the laravel horizon container with PHP 8.1 an error occurs when it tries to install tokenizer. 

I believe this is the same issue as has already been patched in by #3142 

### Context information:

**Output of `git rev-parse HEAD`**

```
fee2d33ad86133bcc354d4ee856b20ab3b60add0
```

**Output of `docker version`**

```
Client:
 Cloud integration: v1.0.22
 Version:           20.10.13
 API version:       1.41
 Go version:        go1.16.15
 Git commit:        a224086
 Built:             Thu Mar 10 14:08:44 2022
 OS/Arch:           darwin/amd64
 Context:           default
 Experimental:      true

Server: Docker Desktop 4.6.0 (75818)
 Engine:
  Version:          20.10.13
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.15
  Git commit:       906f57f
  Built:            Thu Mar 10 14:06:05 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.5.10
  GitCommit:        2a1d4dbdb2a1030dc5b01e96fb110a9d9f150ecc
 runc:
  Version:          1.0.3
  GitCommit:        v1.0.3-0-gf46b6ba
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Output of `docker-compose version`**

```
Docker Compose version v2.3.3
```

**System info: Mac, Windows or Linux. Include which disto/version**

```
Mac OS 12.3
```

### Steps to reproduce the issue:
<!--- How can we reproduce the issue? -->

1. Specify `PHP_VERSION=8.1`
2. Build laravel-horizon image
3.

### Stacktrace & Additional info:

```
#36 88.64 make: *** No rule to make target '/usr/src/php/ext/tokenizer/Zend/zend_language_parser.y', needed by '/usr/src/php/ext/tokenizer/Zend/zend_language_parser.c'.  Stop.
------
failed to solve: executor failed running [/bin/sh -c pecl channel-update pecl.php.net;     docker-php-ext-install mysqli mbstring pdo pdo_mysql tokenizer xml pcntl]: exit code: 2
make: *** [rebuild] Error 17
```

make: *** No rule to make target '/usr/src/php/ext/tokenizer/Zend/zend_language_parser.y', needed by '/usr/src/php/ext/tokenizer/Zend/zend_language_parser.c'.  Stop.
ERROR: Service 'laravel-horizon' failed to build: The command '/bin/sh -c pecl channel-update pecl.php.net;     docker-php-ext-install mysqli mbstring pdo pdo_mysql tokenizer xml pcntl exif' returned a non-zero code: 2

Got the same issue on linux

--- a/laravel-horizon/Dockerfile
+++ b/laravel-horizon/Dockerfile
@@ -37,7 +37,12 @@ RUN apk --update add wget \
   procps
 
 RUN pecl channel-update pecl.php.net; \
-    docker-php-ext-install mysqli mbstring pdo pdo_mysql tokenizer xml pcntl
+    docker-php-ext-install mysqli mbstring pdo pdo_mysql xml pcntl; \
+    if [ $(php -r "echo PHP_MAJOR_VERSION;") = "8" ] && [ $(php -r "echo PHP_MINOR_VERSION;") = "1" ]; then \
+      php -m | grep -q 'tokenizer'; \
+    else \
+      docker-php-ext-install tokenizer; \
+    fi
 
 # Add a non-root user to help install ffmpeg:
 ARG PUID=1000

